File: csharp/etl/Etl.cs
using System;
using System.Collections.Generic;

public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        Dictionary<string, int> better = new Dictionary<string, int>();
        foreach (int i in old.Keys)
        {
            foreach (string letter in old[i])
            {
                better[letter.ToLower()] = i;
            }
        }
        return better;
    }
}

File: elixir/etl/lib/etl.ex
defmodule ETL do
  @doc """
  Transforms an old Scrabble score system to a new one.

  ## Examples

    iex> ETL.transform(%{1 => ["A", "E"], 2 => ["D", "G"]})
    %{"a" => 1, "d" => 2, "e" => 1, "g" => 2}
  """
  @spec transform(map) :: map
  def transform(input) do
    input
    |> Map.to_list()
    |> Enum.map(&scoreListPairToTupleList/1)
    |> Enum.concat()
    |> Enum.into(%{})
  end

  defp scoreListPairToTupleList({score, ls}) do
    ls |> Enum.map(fn c -> {String.downcase(c), score} end)
  end

end
